frontend/src/components/Patchnotes.jsx

```diff
@@ -1,14 +1,28 @@
 import React, { useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
 
-const Patchnotes = () => {
-    const { year, month } = useParams();
-    const [patchDetails, setPatchDetails] = useState({ tank: [], damage: [], support: [], mapUpdates: [], bugFixes: [] });
+const Patchnotes = ({ game }) => {
+    const { year, month, version } = useParams();
+    const [patchDetails, setPatchDetails] = useState({
+        tank: [],
+        damage: [],
+        support: [],
+        agentUpdates: [],
+        mapUpdates: [],
+        bugFixes: []
+    });
 
     useEffect(() => {
         const fetchPatchDetails = async () => {
             try {
-                const response = await fetch(`http://localhost:3000/patchnotes/${year}/${month}`);
+                let url = '';
+                if (game === 'overwatch') {
+                    url = `http://localhost:3000/patchnotes/overwatch/${year}/${month}`;
+                } else {
+                    url = `http://localhost:3000/patchnotes/valorant/${version}`;
+                }
```

Comment:
if we just create a separate page component for each game, we can avoid big conditional branches inside useEffect